tests/Http/Controllers/CelebTest.php

```diff
@@ -2,175 +2,414 @@
+            'photo' => 'https://genericPhotos.com'
+        ]);
     }
 
     /** @test */
-    public function guest_cannot_access_celebs_store()
+    public function guest_cannot_create_a_celeb()
     {
-        $this->postJson('celebs')
-            ->assertStatus(403);
+        $this->postJson('celebs', [
+            'name' => 'Joe Generic',
+            'date_of_birth' => Carbon::now()->subYears(30),
+            'photo' => 'https://genericPhotos.com'
+        ])->assertRedirect();
+
+        $this->assertDatabaseMissing('celebs', [
+            'name' => 'Joe Generic',
+            'date_of_birth' => Carbon::now()->subYears(30),
+            'photo' => 'https://genericPhotos.com'
+        ]);
     }
 
     /** @test */
     public function admin_can_see_celebs_edit_view()
     {
-        $celeb = \App\Models\Celeb::factory()->create();
-        $admin = \App\Models\User::factory()->create([
-            'is_admin' => true
-        ]);
+        $celeb = Celeb::factory()->create();
+        $admin = User::factory()->admin()->create();
 
         $this->actingAs($admin)
             ->getJson("celebs/$celeb->id/edit")
             ->assertOk();
     }
 
     /** @test */
-    public function normal_user_cannot_see_celebs_edit_view()
+    public function user_cannot_see_celebs_edit_view()
     {
-        $celeb = \App\Models\Celeb::factory()->create();
-        $user = \App\Models\User::factory()->create([
+        $celeb = Celeb::factory()->create();
+        $user = User::factory()->create([
             'is_admin' => false
         ]);
 
         $this->actingAs($user)
             ->getJson("celebs/$celeb->id/edit")
-            ->assertStatus(403);
+            ->assertRedirect();
     }
 
     /** @test */
     public function guest_cannot_see_celebs_edit_view()
     {
-        $celeb = \App\Models\Celeb::factory()->create();
+        $celeb = Celeb::factory()->create();
         $this->getJson("celebs/$celeb->id/edit")
-            ->assertStatus(403);
+            ->assertRedirect();
     }
 
     /** @test */
-    public function admin_can_access_celebs_update()
+    public function admin_can_update_a_celeb()
     {
-        $celeb = \App\Models\Celeb::factory()->create();
-        $admin = \App\Models\User::factory()->create([
-            'is_admin' => true
+        $celeb = Celeb::factory()->create([
+            'name' => 'Jennifer Fallinbury',
+            'date_of_birth' => Carbon::now()->subYears(30),
+            'photo' => 'https://generic/photo1_700x600'
         ]);
+        $admin = User::factory()->admin()->create();
 
         $this->actingAs($admin)
-            ->putJson("celebs/$celeb->id")
-            ->assertOk();
+            ->put("celebs/$celeb->id", [
+                'name' => 'Jennifer Jaxon',
+                'date_of_birth' => Carbon::now()->subYears(30),
+                'photo' => 'https://generic/photo1_700x600'
+            ])->assertSessionHasNoErrors();
```

Comment:
I believe this should assert ok instead of assert session has no errors.

Reply:
Hmm this was giving me a 302 status code apparently (I have now changed it to assertStatus(302))